crates/scripts-tests/src/tests/stake_lock/stake_lock.rs

```diff
@@ -0,0 +1,399 @@
+        .hash_type(gw_types::core::ScriptHashType::Data.into())
+        .build();
+    let chain = setup_chain(&rollup_type_script);
+    // create rollup_cell input
+    let capacity = 1000_00000000u64;
+    let rollup_cell_out_point = random_out_point();
+    let rollup_cell = CellOutput::new_builder()
+        .lock(always_success_script())
+        .type_(Some(state_validator_script()).pack())
+        .capacity(capacity.pack())
+        .build();
+    let global_state = chain.local_state.last_global_state();
+    data_loader.cells.insert(
+        rollup_cell_out_point.clone(),
+        (rollup_cell.clone(), global_state.as_bytes()),
+    );
+    let rollup_cell_input = CellInput::new_builder()
+        .previous_output(rollup_cell_out_point)
+        .build();
+
+    let rollup_cell_output = rollup_cell;
+    let mem_pool_package = {
+        let param = PackageParam {
+            deposition_requests: Vec::new(),
+            max_withdrawal_capacity: std::u128::MAX,
+        };
+        chain.mem_pool.lock().package(param).unwrap()
+    };
+    let param = ProduceBlockParam {
+        block_producer_id: 0,
+    };
+    let block_result = chain.produce_block(param, mem_pool_package).unwrap();
+    let rollup_cell_data = block_result.global_state.as_bytes();
+    let witness = WitnessArgs::new_builder()
+        .output_type(Some(block_result.block.as_bytes()).pack())
+        .build();
+
+    // create stake_lock input
+    let capacity = 1000_00000000u64;
+    let owner_lock_hash = always_success_script().calc_script_hash();
+    let rollup_type_hash = state_validator_script().calc_script_hash();
+    let stake_lock_script = stake_lock_script(rollup_type_hash, 100u64, owner_lock_hash);
+    let stake_cell = CellOutput::new_builder()
+        .lock(stake_lock_script)
+        .capacity(capacity.pack())
+        .build();
+    let stake_cell_out_point = random_out_point();
+    data_loader.cells.insert(
+        stake_cell_out_point.clone(),
+        (stake_cell.clone(), Default::default()),
+    );
+    let stake_cell_input = CellInput::new_builder()
+        .previous_output(stake_cell_out_point)
+        .build();
+    // create stake_lock output
+    let stake_cell_output = stake_cell;
+
+    // create tx
+    let tx = Transaction::default()
+        .as_advanced_builder()
+        .cell_dep(state_validator_dep)
+        .cell_dep(stake_lock_dep)
+        .input(rollup_cell_input)
+        .input(stake_cell_input)
+        .output(rollup_cell_output)
+        .output(stake_cell_output)
+        .output_data(rollup_cell_data.pack())
+        .build();
+    let resolved_tx = build_resolved_tx(&data_loader, &tx);
+    let mut verifier = TransactionScriptsVerifier::new(&resolved_tx, &data_loader);
+    verifier.set_debug_printer(|_script, msg| println!("[script debug] {}", msg));
+    // TODO: use the dummy stake_validator to test
+    assert_eq!(verifier.verify(MAX_CYCLES).is_err(), true);
```

Comment:
Check the specific exit number.

Reply:
And I notice this test is wrong. Please don't leave a wrong test case in the code.

Reply:
fixed.